drv/transceivers-server/src/main.rs

```diff
@@ -378,9 +380,11 @@ impl ServerImpl {
                     }
                 }
             } else if !operational && self.thermal_models[i].is_some() {
-                // This transceiver went away; remove it from the thermal loop
-                if let Err(e) = self.thermal_api.remove_dynamic_input(i) {
-                    ringbuf_entry!(Trace::ThermalError(i, e));
+                if let Some(thermal_api) = &self.thermal_api {
+                    // This transceiver went away; remove it from the thermal loop
+                    if let Err(e) = thermal_api.remove_dynamic_input(i) {
+                        ringbuf_entry!(Trace::ThermalError(i, e));
+                    }
```

Comment:
following my suggestion from above:
```suggestion
                #[cfg(feature = "thermal-control")]
                {
                    // This transceiver went away; remove it from the thermal loop
                    if let Err(e) = self.thermal_api.remove_dynamic_input(i) {
                        ringbuf_entry!(Trace::ThermalError(i, e));
                    }
```